mm2src/coins/utxo/utxo_common.rs

```diff
@@ -3114,11 +3125,14 @@ pub fn validate_payment<T: UtxoCommonOps>(
                 Ok(t) => t,
                 Err(e) => {
                     if attempts > 2 {
-                        return ERR!(
-                            "Got error {:?} after 3 attempts of getting tx {:?} from RPC",
-                            e,
-                            tx.tx_hash()
-                        );
+                        return MmError::err(ValidatePaymentError::WrongPaymentTx(
+                            UtxoRpcError::Internal(format!(
+                                "Got error {:?} after 3 attempts of getting tx {:?} from RPC",
+                                e,
+                                tx.tx_hash(),
+                            ))
+                            .to_string(),
+                        ));
```

Comment:
This doesn't make sense.
Use can use `e` variable got above and just return
```rust
return MmError::err(ValidatePaymentError::from(e));
```

Reply:
yeah sorry ...`MmError::err(ValidatePaymentError::from(e.into_inner()))`